scripts/deployer.ts

```diff
@@ -0,0 +1,280 @@
+  deployed['contracts']['staking']['block'] = contractData.blocks.staking;
+  deployed['contracts']['priceOracle']['block'] = contractData.blocks.priceOracle;
+
+  //set abi and bytecode
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const postageABI = await require('../artifacts/src/PostageStamp.sol/PostageStamp.json');
+  deployed['contracts']['postageStamp']['abi'] = postageABI.abi;
+  deployed['contracts']['postageStamp']['bytecode'] = postageABI.bytecode.toString();
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const redisABI = await require('../artifacts/src/Redistribution.sol/Redistribution.json');
+  deployed['contracts']['redistribution']['abi'] = redisABI.abi;
+  deployed['contracts']['redistribution']['bytecode'] = redisABI.bytecode.toString();
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const stakingABI = await require('../artifacts/src/Staking.sol/StakeRegistry.json');
+  deployed['contracts']['staking']['abi'] = stakingABI.abi;
+  deployed['contracts']['staking']['bytecode'] = stakingABI.bytecode.toString();
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const oracleABI = await require('../artifacts/src/PriceOracle.sol/PriceOracle.json');
+  deployed['contracts']['priceOracle']['abi'] = oracleABI.abi;
+  deployed['contracts']['priceOracle']['bytecode'] = oracleABI.bytecode.toString();
+
+  // Construct URL for contract
+  let urlAddress = '';
+  let fileName = '';
+  switch (blockChainVendor) {
+    case 'testnet':
+      urlAddress = testnetURL;
+      fileName = 'testnet_deployed.json';
+      break;
+    case 'mainnet':
+      urlAddress = mainnetURL;
+      fileName = 'mainnet_deployed.json';
+      break;
+    default:
+      break;
+  }
+  if (urlAddress.length != 0) {
+    deployed['contracts']['postageStamp']['url'] = urlAddress + contractData.addresses.postageStamp;
+    deployed['contracts']['redistribution']['url'] = urlAddress + contractData.addresses.redistribution;
+    deployed['contracts']['staking']['url'] = urlAddress + contractData.addresses.staking;
+    deployed['contracts']['priceOracle']['url'] = urlAddress + contractData.addresses.priceOracle;
+  } else {
+    deployed['contracts']['postageStamp']['url'] = urlAddress;
+    deployed['contracts']['redistribution']['url'] = urlAddress;
+    deployed['contracts']['staking']['url'] = urlAddress;
+    deployed['contracts']['priceOracle']['url'] = urlAddress;
+  }
+
+  if (fileName.length == 0 || !fs.existsSync(fileName)) {
+    fileName = blockChainVendor + '_deployed.json';
+  }
+  fs.writeFileSync(fileName, JSON.stringify(deployed, null, '\t'));
+}
+
+// We recommend this pattern to be able to use async/await everywhere
+// and properly handle errors.
+main()
+  .then(() => process.exit(0))
+  .catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
+
+function handleErr(err: string) {
```

Comment:
Honestly, I would suggest simply throwing an `Error` instead of this way of handling it. 

It has benefits like you see stack trace and IMHO is cleaner from readability as well, as you know what happens. Moreover, this exact handling is then performed on the top-level error handler [here](https://github.com/ethersphere/storage-incentives/pull/59/files#diff-21e04f146ffa17a95150f99a1ff0da362398c106a1a4b029a9e8dc4666113ec6R272)